Feature/weatherapp

NewIncome opened the pull request:
This branch creates a weather app with the following functionalities:
- searches the weather by city name
- gives an error if city not found
- lets you swap from °C to °F
- shows an image of the flag of the country
- displays the current temperate, temperature feel, max and min temperatures
- also displays the date, the wind speed and the humidity
- it displays a picture of the city if found or else a random city image
- lastly it changes the color of the background depending on the temperature (only 4 ranges)

NewIncome committed:
Add webpack-dev-server and test

NewIncome committed:
Add basic html Layout

NewIncome committed:
Add dependencies for css and scss usage from js

- add sass, sass-loader, css-loader and style-loader
- use them in the webpack.config

NewIncome committed:
Add styles

- add styles/style.scss dir & file
- Add basic styling with default background image
All basic styling done!

NewIncome committed:
Fixed styles, add basic js for weather API data get

NewIncome committed:
Show sky-cloud status

- add html elemts to show clouds status
- add styles for clouds and flag
- add styles to compensate on temp side

NewIncome committed:
Add flag moving effect

NewIncome committed:
Create new files for JS sectioned code

- make controller file
- make api-data file
- make dom-work file
- make layout file
Up to here working ok, still an error with onClick on console.

NewIncome committed:
Fix JS error and add all element calls

- modify classes of elements for better call
- fix order of render vs elems obj to FIX ERROR
- modify css class for change in name

NewIncome committed:
Add updateInfo function

- add an object returning the values of a the city search
- call the html elements to be modified in the updateInfo fn
- call the new method in the controller
- modify the button in layout to not refresh the page
Button not working yet.

NewIncome committed:
Fix for button not working

- error was in the elms obj being declared in the beginning so
all the elements were null.
- fix was turning it into a function so it could load values
after being called.

NewIncome committed:
Fix for call to API not working right
- pending fix for values not in correct place

NewIncome committed:
Fix layout and updateInfo fn
- fix for correct values and correct syntax
- new makeNewDate fn to get the actual date

NewIncome committed:
Add the Celsius-Farenheight swap functionality

- add separations in the layout for the degreesymbols in the
temperatures
- add the changTemp function
- add the addetional helper functions for the changeTemp fn
- add the call for the changeTemp function
- add the separate styles for the button change from C° to F°

NewIncome committed:
Fix bad look on temperature degree symbols, in layout and scss

NewIncome committed:
Add background-image change

- add image API method and file
- add method to select a random image in case of no city image
- add code to modify the background
- remove basic css background image, was making an error

NewIncome committed:
- Add pixabay icon ans styles
- Add colorsTemp object for background use
- Add temp reset for units button

NewIncome committed:
Add file for testing GitHub Pages, live link

NewIncome committed:
- Add default background image
- Fix position of pixabay icon
- Fix max-size of cloud icon

NewIncome committed:
Add automatic color background change depending on temperature

NewIncome committed:
Fix for enter key not reloading page when trying to search

NewIncome committed:
Add error message div and styles
Pending err functionality on bad search

NewIncome committed:
Add error for city not found

- add html elements for error message
- add syles for error message
- add js code to show error messsage
Working fine in firefox

NewIncome committed:
Add icons

- add icons for wind-speed and humidity
- add fix for bad look of search button in chrome

NewIncome committed:
Fix linter errors and fix cloud contrst for better color view

NewIncome committed:
Fix scss linter errors

NewIncome committed:
Add autoprefixer to webpack for cross browser compatibility
- add postcss.config.js
- remove comment line in scss file

wathigo commented on src/javascript/dom-work.js:
```diff
@@ -0,0 +1,124 @@
+  return `${Months[date.getMonth() + 1]} ${date.getDate()}, ${date.getFullYear()}`;
+};
+
+const updateInfo = (values, city) => {
```
We can use objtect destructuring to get all the items in values object. For example, we can get
```js
const { TempDT, countryDt } = values;
```
We can then access `TempDT` and `countryDt` directly without using repeating `values `.
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/destructuring_requirement.png) from Pathwright).

wathigo commented on package.json:
```diff
@@ -6,7 +6,8 @@
     "build": "webpack",
-    "watch": "webpack --watch"
+    "watch": "webpack --watch",
+    "server": "webpack-dev-server"
```
**OPTIONAL**
Let's change `"server": "webpack-dev-server"` to `"start": "webpack-dev-server"` to get rid of an error raised on an attempt to start the server.

wathigo commented on src/javascript/api-data.js:
```diff
@@ -0,0 +1,21 @@
+async function getTemp(newCity = 'Toronto') {
+  const weatherData = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${newCity}&units=metric&appid=550a45bc4aa9613e4687926216e69765`)
+    .then((response) => response.json()).then((data) => data);
```
**REQUIRED**
Let's add error handling to handle the case when an error is thrown while processing requests. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

wathigo commented on src/javascript/app-controller.js:
```diff
@@ -0,0 +1,15 @@
+  changeTemp,
+} from './dom-work';
+
+function start() {
```
**REQUIRED**
We should strictly use ES6 syntax. Let's replace function with syntax with and arrow function. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) from Pathwright)

wathigo reviewed (changes requested):
### Changes Required.
Great work guys :clap: 
Let's make a couple of changes before we move on.
Check the comments :point_down: 

**After making the changes, submit another request for review.**

**Happy Coding :heavy_check_mark: 
Simon Wathigo**

NewIncome committed:
Review fixes

- use destructuring to get the newTemp values
- change server word call for webpack-dev-server
- add error handling directly in weatherAPI call
- modified start function definition to use ES6 syntax

NewIncome committed:
Fill Readme file

NewIncome committed:
Fix date month error

stratospherique reviewed (approved):
### Great work implementing the suggested changes :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**
